src/annotator/anchoring/xpath-util.js

```diff
@@ -50,58 +53,35 @@ function getNodePosition(node) {
   return pos;
 }
 
-/**
- * A simple XPath evaluator using jQuery which can evaluate queries of
- *
- * @deprecated
- */
-export function simpleXPathJQuery(nodes, relativeRoot) {
-  const paths = nodes.map((index, node) => {
-    let path = '';
-    let elem = node;
-    while (elem.nodeType === Node.ELEMENT_NODE && elem !== relativeRoot) {
-      let tagName = elem.tagName.replace(':', '\\:');
-      let idx = $(elem.parentNode).children(tagName).index(elem) + 1;
-      path = '/' + elem.tagName.toLowerCase() + `[${idx}]` + path;
-      elem = elem.parentNode;
-    }
-    return path;
-  });
-  return paths.get();
-}
-
 function getPathSegment(node) {
   const name = getNodeName(node);
   const pos = getNodePosition(node);
   return `${name}[${pos}]`;
 }
 
 /**
- * A simple XPath evaluator using only standard DOM methods which can
- * evaluate queries of the form /tag[index]/tag[index].
+ * A simple XPath generator using which can generate XPaths of the form
+ * /tag[index]/tag[index].
+ *
+ * @param {Node} node - The node to generate a path to
+ * @param {Node} root - Root node to which the returned path is relative
  */
-export function simpleXPathPure(nodes, relativeRoot) {
-  let rootNode = relativeRoot;
+export function xpathFromNode(node, root) {
+  let xpath = '';
 
-  function getPathTo(node) {
-    let xpath = '';
-    let elem = node;
-    while (elem !== rootNode) {
-      if (!elem) {
-        throw new Error(
-          'Called getPathTo on a node which was not a descendant of @rootNode. ' +
-            rootNode
-        );
-      }
-      xpath = getPathSegment(elem) + '/' + xpath;
-      elem = elem.parentNode;
+  /** @type {Node|null} */
+  let elem = node;
+  while (elem !== root) {
+    if (!elem) {
+      throw new Error('Node is not a descendant of root');
     }
-    xpath = '/' + xpath;
-    xpath = xpath.replace(/\/$/, '');
-    return xpath;
+    xpath = getPathSegment(elem) + '/' + xpath;
+    elem = elem.parentNode;
   }
-  const paths = nodes.map((index, node) => getPathTo(node));
-  return paths.get();
+  xpath = '/' + xpath;
+  xpath = xpath.replace(/\/$/, '');
```

Comment:
There was not a comment here before but this it warrants one since this regex is a bit confusing.

something like "Removes a trailing slash"